src/test/java/teammates/test/driver/PriorityInterceptor.java

```diff
@@ -34,86 +34,82 @@
-                int result = 0;
-                Method method = mi.getMethod().getMethod();
-                Priority a1 = method.getAnnotation(Priority.class);
-                if (a1 != null) {
-                    result = a1.value();
-                }
-                return result;
-            }
-
-            private int getClassPriority(IMethodInstance mi) {
-                int result = 0;
-                Method method = mi.getMethod().getMethod();
-                Class<?> cls = method.getDeclaringClass();
-                Priority classPriority = cls.getAnnotation(Priority.class);
-                if (classPriority != null) {
-                    result = classPriority.value();
-                }
-                return result;
-            }
-
-            private String getPackageName(IMethodInstance mi) {
-                return mi.getMethod().getMethod().getDeclaringClass().getPackage().getName();
-            }
-
-            private String getClassName(IMethodInstance mi) {
-                return mi.getMethod().getMethod().getDeclaringClass().getName();
-            }
-
-            private int packagePriorityOffset(String packageName) {
-                int index = packageOrder.indexOf(packageName);
-
-                if (index == -1) {
-                    return 0;
-                }
-                return -index;
-            }
-
-            @Override
-            public int compare(IMethodInstance m1, IMethodInstance m2) {
-                int val = 0;
-
-                //Compare by package name
-                String p1 = getPackageName(m1);
-                String p2 = getPackageName(m2);
-                val = p1.compareTo(p2);
-                val -= packagePriorityOffset(p1);
-                val += packagePriorityOffset(p2);
-                if (val != 0) {
-                    return val;
-                }
-
-                //Compare by class priority
-                val = getClassPriority(m1) - getClassPriority(m2);
-                if (val != 0) {
-                    return val;
-                }
-
-                //Compare by class name
-                val = getClassName(m1).compareTo(getClassName(m2));
-                if (val != 0) {
-                    return val;
-                }
-
-                //Compare by method priority
-                val = getMethodPriority(m1) - getMethodPriority(m2);
-                if (val != 0) {
-                    return val;
-                }
-
-                return 0;
-            }
+                                           ITestContext context) {
+
+        //Compare by package name
+        Comparator<IMethodInstance> compareByPackage = (IMethodInstance m1, IMethodInstance m2) -> {
+            int val = 0;
 
+            String p1 = getPackageName(m1);
+            String p2 = getPackageName(m2);
+            val = p1.compareTo(p2);
+            val -= packagePriorityOffset(p1);
+            val += packagePriorityOffset(p2);
```

Comment:
All the above can be combined; in fact you can remove the usage of an explicit variable `val` that way.